### PR TITLE
Update VRT350 for SW=0114 HW=7102

### PR DESCRIPTION
**Pull request by TeunvandeBerg**
Add DesiredTempDay, Mode, DesiredTempLow, DesiredTemp

Some backwards compatibility to versions other than SW=0114 HW=7102 is
added, however because an OR expression does not appear to be available
backwards compatibility isn't a 100%.

### CODE REVIEW

File: ebusd-2.1.x/de/vaillant/15.350.csv [hunk 1]
```diff
@@ -5,6 +5,7 @@
 *wi,#install,,,,,"B509","0E",,,,,,
 *ws,#service,,,,,"B509","0E",,,,,,
 *[SW],scan,,,SW,,,,,,,,,
+*[HW],scan,,,HW,,,,,,,,,
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/15.350.csv:8]: usually the HW version is irrelevant and only the SW decides about features. so please remove all HW parts

File: ebusd-2.1.x/de/vaillant/15.350.csv [hunk 2]
```diff
@@ -44,16 +45,19 @@ r;wi,,OffsetDesTemp,,,,,"2300",,,temp,,,
 r;wi,,StateOfRoomCon,,,,,"2400",,,UCH,,,
 r;wi,,LegioProtectionState,,,,,"2500",,,onoff,,,
 r;wi,,StatusDcf,,,,,"2600",,,dcfstate,,,
-r;wi,,DcfDaten,,,,,"2700",,,UIN,,,
+[SW<114][HW<7102]r;wi,,DcfDaten,,,,,"2700",,,UIN,,,
+[SW=114][HW=7102]r;wi,,DesiredTempDay,,,,,"2700",,,D1C,,,
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/15.350.csv:49]: usually features do not disappear with newer versions, so use `[SW>=144]` instead here and also below

File: ebusd-2.1.x/en/vaillant/15.350.csv [hunk 3]
```diff
@@ -5,6 +5,7 @@
 *wi,#install,,,,,"B509","0E",,,,,,
 *ws,#service,,,,,"B509","0E",,,,,,
 *[SW],scan,,,SW,,,,,,,,,
+*[HW],scan,,,HW,,,,,,,,,
```
Comment (john30) [inline ebusd-2.1.x/en/vaillant/15.350.csv:8]: do the same thing in the EN file as noted in the DE

**Review by john30 — CHANGES_REQUESTED** [review]

**Comment by GrumpyMeow** [comment]
Thanks Teun. I'll be using your modifications also.

**Comment by TeunvandeBerg** [comment]
> 
> 
> Thanks Teun. I'll be using your modifications also.

Nice to hear that sharing these changes benefits others.

File: ebusd-2.1.x/de/vaillant/15.350.csv [hunk 4]
```diff
@@ -44,16 +44,19 @@ r;wi,,OffsetDesTemp,,,,,"2300",,,temp,,,
...
 r;wi,,IsInParty,,,,,"2900",,,onoff,,,
 r;wi,,IsInSingleHwcLoadingMode,,,,,"2A00",,,onoff,,,
 r;wi,,RestoreOpModeAfterHoliday,,,,,"2B00",,,onoff,,,
 r;wi,,IsInSavingsFunction,,,,,"2C00",,,onoff,,,
 r;wi,,SavingsFunctionTime,,,,,"2D00",,,time;onoff,,,
 r;wi,,SwitchOffParty,,,,,"2E00",,,onoff,,,
-r;wi,,IsInTeleSwitch,,,,,"2F00",,,onoff,,,
-r;wi,,IsInFloorPavingDrying,,,,,"3000",,,onoff,,,
+[SW<114]r;wi,,IsInTeleSwitch,,,,,"2F00",,,onoff,,,
+[SW>=114]r;wi,,Mode,,,,,"2F00",,,BCD,,, # Day=3 Low=4 Auto=2 Summer=5.
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/15.350.csv:57]: is this key/value list right? if so, use the divisor/values column to assign them

File: ebusd-2.1.x/de/vaillant/15.350.csv [hunk 5]
```diff
@@ -65,7 +68,8 @@ r;wi,,eBUSCRC,,,,,"3800",,,UCH,,,
 r;wi,,HwcTempDesired,,,,,"3900",,,temp,,,
 r;wi,,ComfTempEnabled,,,,,"3A00",,,onoff,,,
 r;wi,,ComfTempDesired,,,,,"3B00",,,temp;temp;temp;temp,,,
-r;wi,,FrostProtectDelay,,,,,"3C00",,,UCH,,,
+[SW<114]r;wi,,FrostProtectDelay,,,,,"3C00",,,UCH,,,
+[SW>=114]r;w,,DesiredTemp,,,,,"3C00",,,D1C,,, # Note: Reading and writing is only effective when a specifc temperature for today has already been set on the VRT350 today. DesiredTemp is then used rather than DesiredTempLow/DesiredTempDay. How to activate DesiredTemp remotely?
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/15.350.csv:72]: why don't you put this into the comment column of the field or the message? german would be preferred for the de file

**Review by john30 — CHANGES_REQUESTED** [review]

File: ebusd-2.1.x/de/vaillant/15.350.csv (hunk 4 above)
Comment (TeunvandeBerg) [reply ebusd-2.1.x/de/vaillant/15.350.csv]: Yes, this key value list is right.

I've now reworked this pull request and used the contents of 15.370.csv for 15.150.csv. It was brought to my attention by @wlcrs that [others ](https://fromeijn.nl/connected-vaillant-to-home-assistant/) successfully use the 15.370.csv for their VRT350.

**Comment by TeunvandeBerg** [comment]
Succeeded by https://github.com/john30/ebusd-configuration/pull/253

**Comment by Florian-ops** [comment]
Hi John, 
I hope you find my note. I just added my Vaillant Boiler with VRT 350 to Home Assistant running the ebusd (ebusd 23.2.p20230716) as Addon provided by Lukas Grebe. 

In the log the demon find the VRT 350 and looks up 15.350.csv whereby I assume it shoud be *370.
Log: _[main notice] read scan config file vaillant/15.350.csv for ID "35000", SW0114, HW7102_
For the boiller it finds 08.bai instead of 08.bai.HW7401
Log: _[main notice] read scan config file vaillant/08.bai.csv for ID "bai00", SW0518, HW7401_

Can I set these values directly via SSH?

**Comment by TeunvandeBerg** [comment]
> Hi John, I hope you find my note. I just added my Vaillant Boiler with VRT 350 to Home Assistant running the ebusd (ebusd 23.2.p20230716) as Addon provided by Lukas Grebe.
> 
> In the log the demon find the VRT 350 and looks up 15.350.csv whereby I assume it shoud be *370. Log: _[main notice] read scan config file vaillant/15.350.csv for ID "35000", SW0114, HW7102_ For the boiller it finds 08.bai instead of 08.bai.HW7401 Log: _[main notice] read scan config file vaillant/08.bai.csv for ID "bai00", SW0518, HW7401_
> 
> Can I set these values directly via SSH?

Hi Florian-ops,

Your question isn't fully clear to me. Are inquiring how to change which configuration is loaded? If so, did you see the following section in the wiki: https://github.com/john30/ebusd/wiki/4.7.-Automatic-configuration

Kind regards,
Teun

**Pull request closed without merging**